### DISCUSSION TITLE
New to Sanni, NES question

### DISCUSSION THREAD

**Discussion by PsychoFox11** [sanni/cartreader#705] [Need help]
I've dumped several NES games so far, which have worked fine. I've done about 4-5. Now I'm testing Magmax for the US NES. The Sanni picks it up as Magmax (Japan) and no matter whether I use that or manually select the (USA) version, it shows the CRC32 then after a few seconds 'not found'.

I can't open either in an emulator either. I've cleaned the cart and reseated it numerous times. I also dumped the bin without the headers to search online for its crc32 but found nothing, none of the crc32 values for any of the .nes files or .bin are anywhere. Is this just a bad dump and a pin isn't getting read, what can I do? Part of what I want this for is checking lots of games for rare or undocumented variants, but I doubt my 5th common game ever is special. 

I did search other threads about NES dumping, but didn't find anything covering it showing up as the right game but wrong region and not working on either. I'm open to suggestions or info! Thanks!

**Comment by sanni** [comment]
[Magmax](https://nescartdb.com/profile/view/236/magmax) seems to be a very standard mapper 0 game. Almost the same as [Super Mario Bros.](https://nescartdb.com/profile/view/2122/super-mario-bros), just horizontal instead of vertical mirroring, which could be a clue to why it doesn't work.

The standard recommendation here would be to clean the pins very well. The OSCR is very very picky when it comes to even the slightest bit of "black stuff" on the pins. Even if the game plays fine in a console and the pins look all shiny and golden. Always use a q-tip with rubbing alcohol and go over the gold fingers until the q-tip doesn't turn black anymore.

What you also can do is to open both the bad dump and a know good copy of the ROM in a hex editor like [Nexus HxD](https://mh-nexus.de/en/hxd/) and use the compare function(Control+K) to see how the ROMs differ from each other. Ofc that will only help if there are just small differences, if the files are completely different then something major is going wrong.

And could you post the log file found in the same directory as the ROM? 

Example:
![wgfafsadffafafff](https://user-images.githubusercontent.com/19169494/218257084-35d76fac-3408-4637-b0b9-c686f588a950.jpg)

It could also be a mechanical issue. Are you using HW3 with an adapter or HW5 with the integrated NES slot? There have been [reports](https://github.com/sanni/cartreader/issues/632) that some NES games have the gold fingers start further up the cartridge.

**Reply by PsychoFox11** [reply]
I got it finally. It was odd as I did 20 more, some needed a second try, but most on first try. Cleaning them every time. I cleaned MagMax a little harder than I normally press down on them, to scrub anything that might be harder. I got a tiny more black off it, and then it dumped correctly. It did detect it as the Japan version, but I changed to US and the dump went without an issue, and the CRC32 matches the common known one. I saw this with a few other games, detecting Japan (or maybe simply listing the first one alphabetically), but dumping to US fine. Can it sometimes just not detect the region? If they were simply the same I'd expect it to be World, which some were.

**Reply by sanni** [reply]
The OSCR reads a small part of the cartridge and compares the checksum to the ones in nes.txt. If the Japanese and US version are the same in that small part that gets checked the OSCR needs user input to select the correct version. In this case just turn the rotary encoder left/right to select the correct region.
